app/models/manageiq/providers/hawkular/middleware_manager/middleware_domain.rb

```diff
@@ -0,0 +1,4 @@
+module ManageIQ::Providers
```

Comment:
> perhaps I missed it, but where is acts_as_miq_taggable that enables tagging?

yeah, I know, I've added it in a followup PR
_https://github.com/ManageIQ/manageiq/pull/10172/files#diff-f783050e3d83a5114e9f9733a53ab741R5_

> aren't the new models extending ActiveRecord or I misunderstood something? (like, for example, middleware_deployment, middleware_server, etc.)

They are, here https://github.com/ManageIQ/manageiq/pull/10048/files#diff-f783050e3d83a5114e9f9733a53ab741R1

I don't understand the need for the duplicity either, but I feel it has to do something with the general middleware entity ([link](https://github.com/ManageIQ/manageiq/blob/master/app/models/middleware_server.rb)) X hawkula entity ([link](https://github.com/ManageIQ/manageiq/blob/master/app/models/manageiq/providers/hawkular/middleware_manager/middleware_server.rb))

I used the same approach.. acutally, now I've noticed that 1 is under `app/models/` and the second that extends the first is under `app/models/manageiq/providers/hawkular/middleware_manager/`.